### DISCUSSION TITLE
Offline use possible? (no connection to cloud)

### DISCUSSION THREAD

**Discussion by christianvw** [PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor#292] [Q&A] [answered]
Hi,

first of all: great work, it works very good without much effort!

**One question that came to my mind: Does the map integration work offline?**

Meaning: No connection to the cloud, only vacuum (ip + token)?

As soon as I close the ports 8883 and 443 outgoing in the firewall, the map does not work anymore (blue picture). But in the raw file (.gz) the map is inside (checked with the offline Java application).

Is the offline operation of the integration perhaps not intended / possible or am I doing something wrong?

Nevertheless: Cool integration!

Greetings
Christian

**Comment by PiotrMachowski** [comment, marked as answer]
Map is downloaded from Xiaomi servers, so it won't work offline. If you want to avoid it you have use Valetudo. Map file is there because it was downloaded before you have block Internet access

**Reply by ShinigamiJ** [reply]
Hello Piotr,

Thank you for this integration! It's been two years since your answer, and I'd like to ask if there's any change to the possibility of local control. The Lydsto vacuum I have is not supported by Valetudo, and the Xiaomi Miot allows local control. If it's possible to save a raw map locally, can't this raw local map then be used in the lovelace card and then send a start-room-sweep command via the Miot integration?

Thank you for all your contributions.

**Reply by PiotrMachowski** [reply]
@ShinigamiJ how is it possible to save raw map locally?

**Reply by ShinigamiJ** [reply]
I may have misunderstood the documentation provided with the map extractor, but it does save the map in the /config/tmp folder within HA?

**Reply by PiotrMachowski** [reply]
I can save the map, but it is just a snapshot downloaded from the cloud - it doesn't work locally. If you want to use the snapshot then you can just take a snapshot of the camera image and save the calibration points from camera attributes - it will work perfectly fine with the map card. It should be a lighter approach compared to parsing the file.
